immune_subtype_clustering/prepare_data_sheet.py

```diff
@@ -0,0 +1,99 @@
+import os
+import pandas as pd
+import synapseclient
+from synapseclient import File
+from typing import List
+
+
+def syn_login():
+    syn = synapseclient.Synapse()
+    syn.login()
+    return syn
+
+
+def download_data_files(parent: str):
+    """downloads all TSV files in parent synapse folder from synapse id provided"""
+    syn = syn_login()
+    children = syn.getChildren(parent=parent)
+    syn_id_list = []
+    for child in children:
+        if str(child["name"]).endswith(".tsv"):
+            syn_id_list.append(child["id"])
+            syn.get(str(child["id"]), downloadLocation="./data")
+    return syn_id_list
+
+
+def load_data_files(data_directory: str) -> List[pd.DataFrame]:
+    """loads all files in data_diectory into dataframes, creates Hugo column,
+    grabs Hugo and tpm columns, renames tppm column to patient id, appends each df to gene_df_list
+    """
+    data_directory = "./data"
+    files = [f for f in os.listdir(data_directory)]
+    print(f"Files: {files} retrieved from data_directory")
+    gene_df_list = []
+    for file in files:
+        patient_id = file.split("_")[0]
+        df = pd.read_csv(data_directory + "/" + file, sep="\t")
+        df.insert(0, "Hugo", df["target_id"].str.split("|", expand=True)[5])
```

Comment:
Just to confirm, the Hugo symbol is in the 6th value of the pipe?

`VAL0|VAL1|VAL2|VAL3|VAL4|VAL5|VAL6`

TIL: `expand=True`

Reply:
I believe so. Here is an example:
```
ENST00000456328.2|ENSG00000223972.5|OTTHUMG00000000961.2|OTTHUMT00000362751.1|DDX11L1-002|DDX11L1|1657|processed_transcript|
```